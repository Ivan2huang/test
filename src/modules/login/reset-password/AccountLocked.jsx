import React from 'react';
import * as PropTypes from 'prop-types';
import { compose } from 'redux';
import { injectIntl } from 'react-intl';
import { Box } from '@material-ui/core';

import CONFIG from '../../../constants/config';
import { formatMessage } from '../../../helpers/helpers';
import Typography from '../../../uiComponents/Typography';

const contentAlignment = {
  cendol: 'center',
  basil: 'center',
  default: 'left',
};

const AccountLocked = ({ intl }) => {
  const align = contentAlignment[CONFIG.themeCode] || contentAlignment.default;

  const message = formatMessage(
    intl,
    'accountLocked.header.desc',
    `This account has been locked due to multiple failures during verification attempts. Please contact your customer service team via ${CONFIG.CSEmail} to unlock your account.`,
    {
      CSEmail: CONFIG.CSEmail,
    },
  );

  const title = formatMessage(
    intl,
    'accountLocked.header.title',
    'Contact Customer Support',
  );

  return (
    <>
      <Box display="flex" textAlign={align} justifyContent={align}>
        <Typography type="style-1">{title}</Typography>
      </Box>
      <Box mt={4} display="flex" textAlign={align} justifyContent={align}>
        <Typography type="style-6">{message}</Typography>
      </Box>
    </>
  );
};

AccountLocked.propTypes = {
  intl: PropTypes.shape({}).isRequired,
};

export default compose(injectIntl)(AccountLocked);
