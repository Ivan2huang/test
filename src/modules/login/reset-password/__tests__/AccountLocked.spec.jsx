/* eslint-disable react/prop-types */

import { render } from '@testing-library/react';
import React from 'react';

import AccountLocked from '../AccountLocked';
import withIntl from '../../../../i18n/withIntlProvider';

jest.mock('../../../error/Error', () => props => (
  <div>
    <div>Error Component</div>
    <span>
      {Object.entries(props).map(([key, value]) => `${key} = ${value}`)}
    </span>
  </div>
));

describe('AccountLocked Component', () => {
  const props = {
    messsage: 'message',
  };
  const setUp = (componentProps = {}) => {
    const Component = withIntl(AccountLocked);
    return render(<Component {...componentProps} />);
  };

  it('should match the snapshot', () => {
    const { container } = setUp(props);

    expect(container).toMatchSnapshot();
  });
});
