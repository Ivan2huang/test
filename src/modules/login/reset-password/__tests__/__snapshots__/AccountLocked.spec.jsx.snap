// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountLocked Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiTypography-root makeStyles-root-3 makeStyles-root-4 MuiTypography-body1"
    >
      Contact Customer Support
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-35"
  >
    <div
      class="MuiTypography-root makeStyles-root-36 makeStyles-root-37 MuiTypography-body1"
    >
      This account has been locked due to multiple failures during verification attempts. Please contact your customer service team via undefined to unlock your account.
    </div>
  </div>
</div>
`;
